Dockerfiles/CI-Dockerfile

```diff
@@ -0,0 +1,43 @@
+### Note: This DockerFile is adapted from https://github.com/ucb-bar/chipyard/blob/master/.circleci/images/Dockerfile which was adapted from: https://github.com/CircleCI-Public/example-images/openjdk
```

Comment:
Nit(s): Lets change the folder to `dockerfiles` since everything else in the repo is lowercase. Also by the next Chipyard release (when this gets put on `master`) this comment will be inaccurate. Maybe just delete the reference to `master` and add a hash or just delete the reference to CY `master`.